### PR TITLE
Feature: Log Request Data

### PR DESCRIPTION
**Pull request by ojengwa**
As a trade-off, I added an extra a settings parameter `REQUEST_LOG_DATA` which determines whether the Request payload should be logged. Also, I noticed that Django only allows the request's data stream to be read once. [You cannot access body after reading from request's data stream](https://code.djangoproject.com/ticket/16822). This is mostly caused by middlewares (Django CORs Headers, in my case).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 django-request
 ==============
 
-[![Build Status](http://img.shields.io/travis/kylef/django-request/master.svg?style=flat)](https://travis-ci.org/kylef/django-request)
+[![Build Status](https://travis-ci.org/kylef/django-request.svg?branch=feature-log-request-data)](https://travis-ci.org/kylef/django-request)
```
Comment (ZuluPro) [inline README.md]: You shouldn't change this, after merge it will be incorrect.

**Comment by felixxm** [comment]
@ojengwa Can you rebase from master and squash your commits?

File: request/migrations/0001_initial.py [hunk 2]
```diff
@@ -15,18 +16,30 @@ class Migration(migrations.Migration):
...
-                ('method', models.CharField(default=b'GET', max_length=7, verbose_name='method')),
-                ('path', models.CharField(max_length=255, verbose_name='path')),
-                ('time', models.DateTimeField(auto_now_add=True, verbose_name='time')),
-                ('is_secure', models.BooleanField(default=False, verbose_name='is secure')),
-                ('is_ajax', models.BooleanField(default=False, help_text='Wheather this request was used via javascript.', verbose_name='is ajax')),
+                ('id', models.AutoField(
```
Comment (felixxm) [inline request/migrations/0001_initial.py]: You can not change initial migration. Please add new migration.

File: request/models.py [hunk 3]
```diff
@@ -16,22 +19,32 @@
...
+    is_ajax = models.BooleanField(_('is ajax'), default=False, help_text=_(
+        'Wheather this request was used via javascript.'))
 
     # User infomation
     ip = models.GenericIPAddressField(_('ip address'))
-    user = models.ForeignKey(AUTH_USER_MODEL, blank=True, null=True, verbose_name=_('user'))
-    referer = models.URLField(_('referer'), max_length=255, blank=True, null=True)
-    user_agent = models.CharField(_('user agent'), max_length=255, blank=True, null=True)
-    language = models.CharField(_('language'), max_length=255, blank=True, null=True)
+    user = models.ForeignKey(
```
Comment (felixxm) [inline request/models.py]: Please do not change formatting.

File: setup.py [hunk 4]
```diff
@@ -15,10 +15,11 @@
     As well as a site statistics module, with the active_users template tag and manager method you can also use django-request to show who is online in a certain time.
     """,
     author='Kyle Fuller',
-    author_email='kyle@fuller.li',
+    author_email='inbox@kylefuller.co.uk',
     url=request.__URL__,
     download_url='http://github.com/kylef/django-request/archive/%s.zip' % request.__version__,
-    packages=['request', 'request.migrations', 'request.templatetags', 'request.management', 'request.management.commands'],
+    packages=['request', 'request.migrations', 'request.templatetags',
```
Comment (felixxm) [inline setup.py]: Please do not change formatting.

**Comment by felixxm** [comment]
It's time to close this PR (see #35).

**Pull request closed without merging**